interface/batchcom/settingsnotification.php

```diff
@@ -28,10 +28,12 @@
     if (!$form_err) 
     {
         $sql_text=" ( `SettingsId` , `Send_SMS_Before_Hours` , `Send_Email_Before_Hours` , `SMS_gateway_password` , `SMS_gateway_apikey` , `SMS_gateway_username` , `type` ) ";
-        $sql_value=" ( '".$_POST[SettingsId]."' , '".$_POST[Send_SMS_Before_Hours]."' , '".$_POST[Send_Email_Before_Hours]."' , '".$_POST[SMS_gateway_password]."' , '".$_POST[SMS_gateway_apikey]."' , '".$_POST[SMS_gateway_username]."' , '".$type."' ) ";
+        $sql_value = " (?, ?, ?, ?, ?, ?, ?) ";
+        $values = array($_POST['SettingsId'], $_POST['Send_SMS_Before_Hours'], $_POST['Send_Email_Before_Hours'], $_POST['SMS_gateway_password'], 
+        $_POST['SMS_gateway_apikey'], $_POST['SMS_gateway_username'],$type);
```

Comment:
This should take care of them being parsed appropriately and prevent other parameters too.

Reply:
I think what is left should also be proper validation of the forms maybe after this gets merged.

Reply:
Ok will look into that too